### PR TITLE
Add "Other" as a job role

### PR DESCRIPTION
**Pull request by mjoyce91**
Adds "Other" as a job role in the New Request form. Conditionally displays a text box for the user to enter their role. This field is required if the user selects "Other", otherwise it is optional (and even display in the form).

Screen Shot 2020-04-11 at 12 44 44 PM

### CODE REVIEW

File: ui/src/views/RequestFormView.tsx [hunk 1]
```diff
@@ -297,6 +301,22 @@ const RequestFormView: React.FC<{ user: User }> = ({ user }) => {
                   </Form.Control>
                 </Form.Group>
 
+                {
+                  detailsReq.jobRole === 'Other' &&
+                    <Form.Group controlId="formBasicOtherRole">
+                      <Form.Label>Other Job Role</Form.Label>
+                      <Form.Control
+                        disabled={disabled}
+                        required // Actually only required when detailsReq.jobRole === 'Other'
```
Comment (ConnorDY) [inline ui/src/views/RequestFormView.tsx:320]: Couldn't you do `required={detailsReq.jobRole === 'Other'}`?

Comment (mjoyce91) [reply ui/src/views/RequestFormView.tsx:320]: I could, but this entire block of code is only even rendered if `jobRole === 'Other'`. I'll add for clarity though 👍

Comment (ConnorDY) [reply ui/src/views/RequestFormView.tsx:320]: Ah I missed that. No need!

Comment (mjoyce91) [reply ui/src/views/RequestFormView.tsx]: Already updated 😃

File: ui/src/views/RequestFormView.tsx [hunk 2]
```diff
@@ -307,6 +311,22 @@ const RequestFormView: React.FC<{ user: User }> = ({ user }) => {
                   </Form.Control>
                 </Form.Group>
 
+                {
+                  detailsReq.jobRole === 'Other' &&
+                    <Form.Group controlId="formBasicOtherRole">
+                      <Form.Label>Other Job Role</Form.Label>
+                      <Form.Control
+                        disabled={disabled}
+                        required={detailsReq.jobRole === 'Other'}
```
Comment (ConnorDY) [inline ui/src/views/RequestFormView.tsx:320]: I didn't think it was necessary to add this as this should always be true. Sorry for the misunderstanding!

**Pull request merged**